src/platform/forms-system/test/js/components/FormSignature.unit.spec.jsx

```diff
@@ -0,0 +1,192 @@
+      );
+      expect(getByLabelText('Custom text here')).to.exist;
+    });
+
+    it('should render input input with custom React element label', () => {
+      const customLabel = <span>Custom text here</span>;
+      const { getByLabelText } = render(
+        <FormSignature {...signatureProps} signatureLabel={customLabel} />,
+      );
+      expect(getByLabelText('Custom text here')).to.exist;
+    });
+  });
+
+  describe('certification checkbox', () => {
+    it('should render with default label', () => {
+      const { getByLabelText } = render(<FormSignature {...signatureProps} />);
+      expect(
+        getByLabelText(
+          'I certify the information above is correct and true to the best of my knowledge and belief.',
+        ),
+      ).to.exist;
+    });
+
+    it('should render with custom string label', () => {
+      const { getByLabelText } = render(
+        <FormSignature {...signatureProps} checkboxLabel="LGTM" />,
+      );
+      expect(getByLabelText('LGTM')).to.exist;
+    });
+
+    it('should render with custom React element label', () => {
+      const customLabel = <span>Custom text here</span>;
+      const { getByLabelText } = render(
+        <FormSignature {...signatureProps} checkboxLabel={customLabel} />,
+      );
+      expect(getByLabelText('Custom text here')).to.exist;
+    });
+  });
+
+  describe('validation', () => {
+    it('should require signature and certification', () => {
+      const { getByText } = render(
+        <FormSignature {...signatureProps} required />,
+      );
+      expect(getByText(inputErrorText)).to.exist;
+      expect(getByText(checkboxErrorText)).to.exist;
+    });
+
+    it('should not show validation errors when showErrors is false', () => {
+      const { queryByText } = render(
+        <FormSignature {...signatureProps} showError={false} required />,
+      );
+      expect(queryByText(inputErrorText)).to.not.exist;
+      expect(queryByText(checkboxErrorText)).to.not.exist;
+    });
+
+    it('should dismiss validation errors after resolution', () => {
+      const { queryByText } = render(
+        <FormSignature {...signatureProps} required />,
+      );
+      userEvent.type(queryByText(/Veteran’s full name/), 'Curious George');
+      userEvent.click(
+        queryByText(/I certify the information above is correct/),
+      );
+      expect(queryByText(inputErrorText)).to.not.exist;
+      expect(queryByText(checkboxErrorText)).to.not.exist;
+    });
+
+    it('should perform data validations if present', () => {
+      const validationSpies = [spy(() => 'Example error'), spy()];
+      render(
+        <FormSignature {...signatureProps} validations={validationSpies} />,
+      );
+      expect(validationSpies[0].calledWithExactly('', signatureProps.formData))
+        .to.be.true;
+
+      // Subsequent validation functions don't get run if previous validators
+      // return an error message
+      expect(validationSpies[1].called).to.be.false;
```

Comment:
This seems like something we may want to mention in the  JSdoc comment on the `validations` propType so that teams know what to expect.